definition_validator/validator.cc

```diff
@@ -1096,7 +1110,14 @@ class ValidateWalk {
                 continue;
             }
 
-            auto concreteMethodRef = sym.data(ctx)->findConcreteMethodTransitive(ctx, proto.data(ctx)->name);
+            // Overload signatures all have unique names, so to find the name of the concrete implementation we need to
+            // use the original name, not the unique one.
+            auto protoName = proto.data(ctx)->name;
+            if (protoName.isOverload(ctx)) {
+                protoName = protoName.dataUnique(ctx)->original;
+            }
+
+            auto concreteMethodRef = sym.data(ctx)->findConcreteMethodTransitive(ctx, protoName);
```

Comment:
Rather than solving this here, should we make it so that getAbstractMethods doesn't return overloaded methods?

Or should we maybe explicitly say that we're going to opt out of checks for overloaded abstract methods, or maybe even ban them entirely because Sorbet isn't currently going to do something sensible with them?

Reply:
It's the same problem again with us not propagating information from the overloads back to the <U ...> symbol: if we only ever put the non-overloaded symbols into the result of getAbstractMethods, we'll end up putting symbols in there that aren't currently marked as abstract. Picking the first overload means that we propagate all the correct information, but now have to deal with having an overload present in that list.